### PR TITLE
[Issue #1658]: Agency filter list

### LINKED ISSUE #1658: [Task]: Set up all agencies as filter options
### Summary

- Make sure every agency appears as an option in the agency filter. 
- Get the values from the API docs

- Create a lookup list from the `agencyFilterList` so we can show our user friendly names based on the agency code. Something like this:
`{ "ARPAH" : "All Advanced research PRojects Agency for Health [ARPAH"}`

### Acceptance criteria

_No response_

### PR DESCRIPTION
**Pull request by rylew1**
## Summary
Fixes #1658

## Changes proposed
- Update agency filter list based on spreadsheet
- generate an agency filter lookup that will use the same labels in the `SearchResultsListItem` that the agency filters use

## Context for reviewers
- spreadsheet => https://docs.google.com/spreadsheets/d/1M14vX-kDakn0QHhYkUnIasBnTFUfU73BKXXxAfCxtzQ/edit#gid=1891194558

### CODE REVIEW

File: frontend/src/utils/search/generateAgencyFilterLookup.ts
Comment (andycochran) [inline frontend/src/utils/search/generateAgencyFilterLookup.ts:1]: `AgencyNameLookup`?

Comment (andycochran) [reply frontend/src/utils/search/generateAgencyFilterLookup.ts:1]: …since it's also used for the results, not just for the filter?

**Comment by rylew1** [comment]
@andycochran @crystabelrangel - let me know what labels we want to manually change

File: frontend/src/utils/search/generateAgencyFilterLookup.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { agencyFilterList } from "src/components/search/SearchFilterAccordion/filterJSONLists/agencyFilterList";
```
Comment (andycochran) [inline frontend/src/utils/search/generateAgencyFilterLookup.ts:1]: You do the list in a separate PR? (Is it mostly there? Looks like it's just commented out.)

Comment (andycochran) [reply frontend/src/utils/search/generateAgencyFilterLookup.ts:1]: Heh. Never mind. You just pushed more.

**Comment by andycochran** [comment]
Can you swap the square brackets for parentheses, like below? 

We'll need to change any of these that don't match. These are the only ones that were reviewed. @crystabelrangel, do you remember which ones we changed during approvals? "ED" came to mind, but it looks like it's already right in the data. 

image

**Review by andycochran — APPROVED** [review]

File: frontend/src/utils/search/generateAgencyFilterLookup.ts (hunk 1 above)
Comment (rylew1) [reply frontend/src/utils/search/generateAgencyFilterLookup.ts:1]: Pardon me -  I was probably tagging folks to review before the final push

File: frontend/src/utils/search/generateAgencyFilterLookup.ts
Comment (rylew1) [reply frontend/src/utils/search/generateAgencyFilterLookup.ts:1]: changed to `generateAgencyNameLookup` and `agencyNameLookup` 

Also changed to parentheses.

Comment (rylew1) [reply frontend/src/utils/search/generateAgencyFilterLookup.ts:1]: image

**Comment by andycochran** [comment]
Can you nix the prefix "All"? We think it's unnecessary and implied by the nesting/dropdown (and will test)

File: frontend/src/components/search/SearchResultsListItem.tsx [hunk 2]
```diff
@@ -99,8 +102,11 @@ export default function SearchResultsListItem({
             <div className="grid-col tablet:order-3 overflow-hidden font-body-xs">
               <span className={metadataBorderClasses}>
                 <strong>Agency:</strong>{" "}
-                {opportunity?.summary?.agency_name
-                  ? opportunity?.summary?.agency_name
+                {opportunity?.summary?.agency_name &&
+                opportunity?.summary?.agency_code &&
+                agencyNameLookup
+                  ? // Use same exact label we're using for the agency filter list
+                    agencyNameLookup[opportunity?.summary?.agency_code]
```
Comment (rylew1) [inline frontend/src/components/search/SearchResultsListItem.tsx:109]: @andycochran - this is the key part here - make sure this is what you want. We're displaying the exact checkbox choice label (if it's actually in the `agencyNameLookup` )

**Comment by rylew1** [comment]
@andycochran I've removed the `All` prefix from the labels

**Review by andycochran — APPROVED** [review]
